dts/arm/stm32l476g_disco.dts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2017 Arthur Sfez
+ *
+ * Based on stm32l496g_disco:
+ *
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <st/stm32l476.dtsi>
+
+/ {
+	model = "STMicroelectronics STM32L476G-DISCO board";
+	compatible = "st,stm32l476g-disco", "st,stm32l476";
+
+	chosen {
+		zephyr,console = &usart2;
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+	};
+};
+
+&usart2 {
+	current-speed = <115200>;
+	pinctrl-0 = <&usart2_pins_c>;
```

Comment:
I have a couple of questions regarding dts/pinmux.

Here, I've selected the `&usart2_pins_c` for the USART2. I'm realizing that this is wrong, because the rx/tx pinout defined for `usart2_pins_c` (in `stm32l4-pinctrl.dtsi`) is not the good one for this board (and so it does not match the pins defined in my `pinmux.c` file). So, here, would the good thing to do be to create a new file `stm32l4x6-pinctrl.dtsi` and define there a new value `usart2_pins_d` ?

And so my 2nd question : the dts files are supposed to define the pin definitions, but this is what the pinmux.c file is doing too (and it seems to override the .dts file, else the `hello_world` demo wouldn't work if it used the pins of `usart2_pins_c`). Why do we have to specify pins twice?

I could use some help to make me understand and contribute with a more complete board package support!
Thanks

Reply:
> Here, I've selected the &usart2_pins_c for the USART2. I'm realizing that this is wrong, because the rx/tx pinout defined for usart2_pins_c (in stm32l4-pinctrl.dtsi) is not the good one for this board (and so it does not match the pins defined in my pinmux.c file, ie PD5 & PD6). What's the good thing to do here? Add a new usart2_pins_d to the stm32l4-pinctrl.dtsi ? 

Yes

> But to do so, I need to make sure that ALL stm32l4 SoC are compatible with this change. Is there a quick way to check this?

You don't. Each board select its specific pin configs.

> And so my 2nd question : the pinctrl-dtsi files are supposed to define the pin definitions, but this is what the pinmux.c file is doing too. Shouldn't pinmux.c use defines generated from the dts files?

This is a transition period from c to dts defined pinmuxing. During this time, we should maintain both, so even if only pinmux.c is actually used, please keep dts in sync.

Reply:
Alright, thanks for the thorough explanations, I'll get on it later today!